.github/workflows/npm-publish.yml

```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: 14
+          node-version: 20
```

Comment:
Thanks!
I realised that I have forgotten another couple of things, can you please align it with [this one](https://github.com/inveniosoftware/react-searchkit/blob/master/.github/workflows/npm-publish.yml)?

Missing:
1. 
```
permissions:
  id-token: write
  contents: read
```
2. upgrade `actions/checkout@v4` to `v5`
3. upgrade `actions/setup-node@v4` to `v6`